### ISSUE TITLE
`insertNewRowBefore` and/or `removeRow`, does not update cell Reference in formula on other sheets

### ISSUE THREAD

**Issue by ngyuki** [PHPOffice/PhpSpreadsheet#3380]
This is:

```
- [x] a bug report
- [ ] a feature request
- [x] **not** a usage question (ask them on https://stackoverflow.com/questions/tagged/phpspreadsheet or https://gitter.im/PHPOffice/PhpSpreadsheet)
```

### What is the expected behavior?

`insertNewRowBefore` and/or `removeRow` will also update cell references in formula on other sheets.

This is same behavior as Excel.

### What is the current behavior?

`insertNewRowBefore` and/or `removeRow` updates only cell references in formula on current sheet.

### What are the steps to reproduce?

**Sheet1**

- A1: `=SUM(Sheet2!A1:A3)`

**Sheet2**

- A1: 1
- A2: 2
- A3: 3

```php
require __DIR__ . '/vendor/autoload.php';

use PhpOffice\PhpSpreadsheet\IOFactory;
use PhpOffice\PhpSpreadsheet\ReferenceHelper;

$book = IOFactory::load(__DIR__ . '/input.xlsx');
$sheet2 = $book->getSheetByName('Sheet2');
$sheet2->removeRow(2);

IOFactory::createWriter($book, IOFactory::WRITER_XLSX)->save(__DIR__ . '/output.xlsx');
```

I would expect the formula for Sheet1!A1 to be updated to `=SUM(Sheet2!A1:A2)`, but actually it will not be updated.

### What features do you think are causing the issue

- [ ] Reader
- [ ] Writer
- [ ] Styles
- [ ] Data Validations
- [ ] Formula Calculations
- [ ] Charts
- [ ] AutoFilter
- [ ] Form Elements

### Does an issue affect all spreadsheet file formats? If not, which formats are affected?

Maybe all file formats.

### Which versions of PhpSpreadsheet and PHP are affected?

php 8.2.2
phpoffice/phpspreadsheet 1.27.1

**Comment by MarkBaker** [comment]
You are correct: this was an early trade-off, because of the high performance cost of updating formulae on insert/delete, and the fact that most people weren't building spreadsheets with formulae that referenced other worksheets, it was decided to limit updates to the one worksheet.

Perhaps it's time to review that original decision

**Comment by franz-josef-kaiser** [comment]
Please see #2509 for already existing performance issues.

**Comment by oleibman** [comment]
This is being tracked in issue #4379. Note that it is somewhat in conflict with issue #2509.
